packages/functions/src/dal/migrationScripts/ChangeAppointmentStatusMigration.ts

```diff
@@ -0,0 +1,39 @@
+import { getAllLatestAppointments, setAppointment } from "../AppointmentDataAccessLayer";
+import { AppointmentStatus } from "@zm-blood-components/common/src";
+const fs = require('fs');
+
+export async function addStatusForAppointments(){
+  const appointments = await getAllLatestAppointments();
+
+  return appointments.map((appointment) => {
+    let newStatus = AppointmentStatus.AVAILABLE;
+    const now = new Date();
+    if(appointment.donorId !== "")
+    {
+      if (appointment.donationStartTime.toDate() < now)
+        newStatus = AppointmentStatus.COMPLETED; // should this be completed?
+      else
+        newStatus = AppointmentStatus.BOOKED;
+    }
+    else
+    {
+      if(appointment.donationStartTime.toDate() < now)
+        newStatus = AppointmentStatus.NOSHOW
+    }
+    return {
+      status: newStatus,
+      ...appointment
```

Comment:
You should switch these 2 lines.

Otherwise if the appointment already has a status it won't be updated

Or is it inetntional?

Reply:
it was intentional.... it was a fallback that if there was already a status it will take the status...

Reply:
Cool